### PR TITLE
Feature/oidc client integration

### LINKED ISSUE #98: Add proper authentication functionality
#### Proposed solution

On top of the authentication outlines set in the authentication middleware, couple things are lacking:
- OAuth2 server(s) should be determined that return a public key for JWT Bearer token validation
- audience claim that states the intended recipient of the token needs to be verified if one exists and is required
- authenticated bearer token issuers need to be included in payload claims

#### DoD (Definition of Done)

- public key is received from OAuth2 server(s)
- audience claim(s) are verified and included in `claims_options`
- authenticated bearer token issuers are referenced in `claims_options['iss']['values']` 

#### Testing

Possibility of integration testing

### PR DESCRIPTION
**Pull request by blankdots**
### Description

This would resolve authenticating via an OIDC endpoint. however to make use of this in a proper way #135 is needed
More testing will come after #135 and #106 

### Related issues

Closes #98 

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Changes Made

- Remove jwt auth as not needed anymore, used OIDC endpoint should be able to handle multiple auth types.
- added auth routes
- added middleware to check login when use authenticates.
- add configuration parameters.
- use 1 worker as session handling is problematic across workers

### Testing

- [x] Unit Tests
- [x] Needs testing (start an issue or do a follow up PR about it)

### Mentions

works with: https://github.com/CSCfi/metadata-submitter-frontend/pull/80

### CODE REVIEW

File: metadata_backend/conf/conf.py [hunk 1]
```diff
@@ -89,3 +89,21 @@ def create_db_client() -> AsyncIOMotorClient:
...
+
+aai_config = {
+    "client_id": os.getenv("CSC_AAI_CLIENT_ID", "public"),
+    "client_secret": os.getenv("CSC_AAI_CLIENT_SECRET", "secret"),
+    "domain": os.getenv("BASE_URL", "http://localhost:5430"),
+    "scope": "openid profile email",
+    "iss": os.getenv("AUTH_URL", ""),
+    "aud": "aud1",
+    "callback_url": f'{os.getenv("BASE_URL", "http://localhost:5430")}/callback',
+    "auth_url": f'{os.getenv("OIDC_URL", "")}/authorize',
```
Comment (csc-jm) [inline metadata_backend/conf/conf.py:104]: When I was testing this branch with the `feature/login-link` branch of the frontend repo, I ran into an error: `Oops! Page /authorize could not be found.`

I assume it's caused by this and that this should instead be:
```suggestion
    "auth_url": f'{os.getenv("OIDC_URL", "")}/aai',
```

**Review by csc-jm — CHANGES_REQUESTED** [review]
Otherwise everything looks great apart from that one thing.

File: metadata_backend/conf/conf.py (hunk 1 above)
Comment (blankdots) [reply metadata_backend/conf/conf.py:104]: nope, that is correct. that is oidc endpoint configuration 
I suspect you were missing something from configuration in docker-compose or did not run the frontend at same url as backend. You need to build and start the full image not the dev one.
In the `Dockerfile` you need to `RUN git checkout feature/login-link` after line 9 to build the image with that branch

**Review by csc-jm — APPROVED** [review]
Great PR

**Pull request merged**